src/rgw/rgw-restore-bucket-index

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+# version 2023-01-11
+
+# IMPORTANT: affects order produced by 'sort' and 'ceph-diff-sorted'
+# relies on this ordering
+export LC_ALL=C
+
+bkt_entry=/tmp/bkt-entry.$$
+bkt_inst=/tmp/bkt-inst.$$
+obj_list=/tmp/object-list.$$
+export data_pools=/tmp/data-pools.$$
+
+log() {
+  echo $(date +%F\ %T) $(hostname -s) "$1"
+}
+
+usage() {
+  >&2 cat << EOF
+
+Usage: $0 <bucket-name> [data-pool-name]
+EOF
+  exit 1
+}
+
+get_field() {
+  grep "\"$2\":" $1 | sed -E "s/.*\".*\": ([^,]*),?/\1/"
+}
+
+# quotes
+get_field_q() {
+  grep "\"$2\":" $1 | sed -E "s/.*\".*\": \"(.*)\".*/\1/"
+}
+
+get_pool() {
+  if [ -n "$1" ] ;then
+    echo "$1"
+  else
+    radosgw-admin zone get >$data_pools 2>/dev/null
+    pool_count=$(grep "\"data_pool\":" $data_pools | wc -l)
+    if [ "$pool_count" -ne 1 ] ;then
+      echo "Error: unable to determine data pool; please specify on command-line."
+      usage
```

Comment:
if you want to get clever about this part, you can determine the correct pool name from the bucket instance metadata's "placement_rule" field

that string will either in the form `"{placement id}/{storage class}"`, or just the placement id if the storage class is the default "STANDARD" one. you'd use those keys to look up the correct data_pool under the zone's placement_pools:
```
    "placement_pools": [
        {
            "key": "default-placement",
            "val": {
                "index_pool": "default.rgw.buckets.index",
                "storage_classes": {
                    "STANDARD": {
                        "data_pool": "default.rgw.buckets.data"
                    }
                },
                "data_extra_pool": "default.rgw.buckets.non-ec",
                "index_type": 0,
                "inline_data": true
            }
```

Reply:
That kind of logic would seem to exceed `sed`'s ability to parse. We'd have to force `jq` to be installed, which Matt thinks is reasonable.

Reply:
I think I've added the appropriate logic to `get_pool()` to cover explicit pools and placement rules. Would you mind taking a look, @cbodley ?

Reply:
(using jq)

Reply:
insofar as i can decipher the awk and jq, it looks like it's doing the right thing